hs_tools_resource/models.py

```diff
@@ -230,9 +234,56 @@ def remove(cls, element_id):
 class ToolIcon(AbstractMetaDataElement):
     term = 'ToolIcon'
     value = models.CharField(max_length=1024, blank=True, default="")
+    data_url = models.TextField(blank=True, default="")
+
+    @classmethod
+    def _validate_tool_icon(cls, url):
+        try:
+            response = requests.get(url, verify=False)
+        except:
+            raise ValidationError("Failed to read data from given url.")
+        if response.status_code != 200:
+            raise ValidationError("Failed to read data from given url.")
```

Comment:
@zhiyuli You should add the status_code and corresponding error message into validation error rather than a uniform ```Failed to read data from given url``` validationError. Or probably better, raise a HttpResponse with status code and error message.